packages/pwa/app/utils/routes-utils.test.js

```diff
@@ -9,38 +9,77 @@ import React from 'react'
 import {configureRoutes} from './routes-utils'
 import {getUrlConfig} from './utils'
 
-jest.mock('./utils', () => {
-    const original = jest.requireActual('./utils')
-    return {
-        ...original,
-        getUrlConfig: jest.fn()
-    }
-})
-
 describe('configureRoutes', function() {
-    test('should only configured the routes with :locale as part of the path', () => {
-        getUrlConfig.mockImplementation(() => ({
-            locale: 'path'
-        }))
-        const CompA = () => <div>This is component A</div>
-        const CompB = () => <div>This is component B</div>
+    const env = process.env
+    beforeEach(() => {
+        jest.resetModules()
+        process.env = {...env}
+    })
+
+    afterEach(() => {
+        process.env = env
+    })
+    const CompA = () => <div>This is component A</div>
+    const CompC = () => <div>This is component C</div>
+
+    const routes = [
+        {
+            path: '/',
+            component: CompA,
+            exact: true
+        },
+        {
+            path: '/category/:categoryId',
+            component: CompC,
+            exact: true
+        }
+    ]
+    test('should return all permutation of path including site and locales ', () => {
+        const configuredRoutes = configureRoutes(routes, getUrlConfig(), {ignoredRoutes: ['/']})
+        expect(configuredRoutes[configuredRoutes.length - 1].path).toEqual('/category/:categoryId')
+        expect(configuredRoutes.length).toEqual(31)
+        const paths = configuredRoutes.map((route) => route.path)
+        expect(paths).toEqual(expectedPathsResult)
+    })
```

Comment:
I'd like to see more tests. Specifically testing different configurations. Is it also not possible to compare arrays, it looks like you are just checking the length and a random index for equality.

Reply:
I've added more tests for different urlConfig [here](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/391/files#diff-527fea249b343fca2c8df71e08569a3a5bb07bbdf2c2956122dc766b9b6ad8fdR12)

Reply:
Thanks. Those new tests cover a lot more than before. I don't see a `site=query_param and locale=query_param` test tho. We might want to cover that as well.

Reply:
I've added that test in.